docs/azure/dependency-injection.md

```diff
@@ -0,0 +1,239 @@
+            .WithName("PrivateStorage");
+    });
+}
+```
+
+In your controllers, you can access the named service clients using the `IAzureClientFactory`:
+
+```csharp
+public class HomeControllers : Controller
+{
+  private BlobServiceClient _publicStorage, _privateStorage;
+
+  public HomeController(BlobServiceClient defaultClient, IAzureClientFactory<BlobServiceClient> clientFactory)
+  {
+    _publicStorage = defaultClient;
+    _privateStorage = clientFactory.GetClient("PrivateStorage");
+  }
+}
+```
+
+The un-named service client is still available in the same way as before. Named clients are additive to this.
+
+## Configure a new retry policy
+
+At some point, you might want to change the default settings for a service client. You may want different retry settings or to use a different service API version, for example. You can set the retry settings globally or on a per service basis. Say you have the following _appsettings.json_ file:
+
+```json
+{
+  "AzureDefaults": {
+    "Retry": {
+      "maxTries": 3
+    }
+  },
+  "KeyVault": {
+    "VaultUri": "https://mykeyvault.vault.azure.net"
+  },
+  "Storage": {
+    "ServiceUri": "https://store1.storage.windows.net"
+  },
+  "CustomStorage": {
+    "ServiceUri": "https://store2.storage.windows.net"
+  }
+  ```
+
+You could change the retry policy depending on your needs like so:
+  
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddAzureClients(builder =>
+    {
+        // Establish the global defaults
+        builder.ConfigureDefaults(Configuration.GetSection("AzureDefaults"));
+        builder.UseCredential(new DefaultAzureCredential());
+    
+        // A Key Vault Secrets client using the global defaults
+        builder.AddSecretClient(Configuration.GetSection("KeyVault"));
+    
+        // A Storage client with a custom retry policy
+        builder.AddBlobServiceClient(Configuration.GetSection("Storage"))
+            .ConfigureOptions(options => options.Retry.MaxRetries = 10);
+    
+        // A named storage client with a different custom retry policy
+        builder.AddBlobServiceClient(Configuration.GetSection("CustomStorage"))
+            .WithName("CustomStorage")
+            .ConfigureOptions(options =>
+            {
+                options.Retry.Mode = Azure.Core.RetryMode.Exponential;
+                options.Retry.MaxRetries = 5;
+                options.Retry.MaxDelay = TimeSpan.FromSections(120);
+            });
+    });
+}
+```
+
+You can also place policy overrides in the _appsettings.json_ file:
+
+```json
+{
+  "KeyVault": {
+    "VaultUri": "https://mykeyvault.vault.azure.net",
+    "Retry": {
+      "maxRetries": 10
+    }
+  }
+}
+```
+
+## See also
+
+- [Dependency injection in ASP.NET Core](https://docs.microsoft.com/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-3.0)
+- [Configuration in ASP.NET Core](https://docs.microsoft.com/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.0)
```

Comment:
@scottaddie and @IEvangelist what do you think of linking to these articles?

Reply:
Great idea! See my suggestion below.

Reply:
I think we should be linking to the .NET specific articles here, both of which do have links to the ASP.NET Core docs if needed. This doc will be used by any workload, not just web apps. As such is should be more broad and encompassing of all .NET workloads where the Azure .NET SDK can be used.

- [Dependency injection in .NET](/dotnet/core/extensions/dependency-injection)
- [Configuration in .NET](/dotnet/core/extensions/configuration)

Reply:
@IEvangelist sorry for the delay. I think that's a good idea